syntax = "proto3";

option csharp_namespace = "ConformanceCommon.Protos";
import "google/protobuf/empty.proto";

service TelemetryModule{
	rpc PostTelemetry (stream Telemetry) returns (google.protobuf.Empty);
}

message Telemetry{
	message PointModel{
		string type = 1;
		double lat = 2;
		double lng = 3;
	}
	enum TrackBearingReference{
		TRUE_NORTH = 0;
		MAGNETIC_NORTH = 1;
	}

    string operation_id = 1;
    string enroute_positions_id = 2;
    string registration_id = 3;
    string reference_number = 4;
    PointModel location = 5;
    double ground_speed_kt = 6;
    int64 time_measured = 7;
    int64 time_send = 8;
    int32 battery_remaining = 9;
    string mode = 10;
    double altitude_ft_wgs84 = 11;
    int32 altitude_num_gps_satellites = 12;
    double hdop_gps = 13;
    double track_bearing_deg = 14;
    TrackBearingReference track_bearing_reference = 15;
    double vdop_gps = 16;
    double roll = 17;
    double yaw = 18;
    double pitch = 19;
    double climbrate = 20;
    double heading = 21;
}